### ISSUE TITLE
Tests for core libraries

### ISSUE THREAD

**Issue by barrysteyn** [serverless/components#511]
# Create tests for core libraries

Can we setup a testing framework (Jest comes to mind, but it can be anything) for the core libraries for components. The tests don't need to be rigorous, but before a PR is accepted, we can insist that the some sort of testing be done. Now I do realize that this project is in Beta, and so again, I am not proposing that we rigorously test things, just setup the framework and provide a modicum of tests that will guide anyone who wishes to contribute.

**Comment by eahefnawy** [comment]
Absolutely! We're doing couple of experiments that might change the API, but once this is set in stone, we'll def add tests.

**Comment by medikoo** [comment]
Best if they're setup with help of [serverless/test](https://github.com/serverless/test) (proven to be bulletproof, and already some other serverless projects rely on this)
